### ISSUE TITLE
[a11y]: Tags outline not focusable

### ISSUE THREAD

**Issue by psurana2** [carbon-design-system/carbon#16197]
### Package

@carbon/react

### Browser

Chrome, Firefox

### Operating System

MacOS

### Package version

v1.50.0

### React version

v17.0.2

### Automated testing tool and ruleset

Manual

### Description

On high contrast OS setting , the tag outlines , focus indicators are not visible on keyboard focus . Also the clickable title lines are not differentiable

### Reproduction/example

https://stackblitz.com/edit/github-bv73gf?file=src%2FApp.jsx

### Steps to reproduce

1. On tab press, the tags are not focusable and the tag outline are not visible 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by psurana2** [comment]
It is a dismissible tag @laurenmrice

**Comment by guidari** [comment]
Hey @psurana2 
The Dismissible tag does not receive focus/clickable. It only receives focus on the close icon to dismiss the tag. So it will not have any outline indicator. 
Same thing for Read Only on your Stackblitz example. If you want a focusable/clickable tag you can use the Operational Tag

**Comment by tw15egan** [comment]
Unable to reproduce in Safari + Chrome with MacOS `Increase Contrast` mode enabled
![2024-04-29 11 59 16](https://github.com/carbon-design-system/carbon/assets/11928039/cfefd061-521b-4447-a5a9-580c99caf145)

If using Safari, ensure the "Press tab to highlight" option is checked in the Advanced tab of the settings panel
